### ISSUE TITLE
PEAR and Squiz FunctionComment sniffs throw errors for some invalid @param line formats

### ISSUE THREAD

**Issue by andykais** [squizlabs/PHP_CodeSniffer#1091]
```
php ~/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar --extensions=php,php3,php4,php5,phtml,inc,lib,module,info --report=xml --standard=PSR2,Generic,Zend,Squiz,PSR1,MySource,PEAR --report-file=phpcs.xml drupal-8.1.x
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 261
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 262
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 261
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 262
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php on line 230
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php on line 231
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 261
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 262
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 261
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php on line 262
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php on line 230
PHP Notice:  Undefined offset: 1 in phar:///home/andrew/Build/codedx/dev/lift-project/bundle/target/scala-2.11/classes/bundled-tools/php-codesniffer-2.6.1/phpcs.phar/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php on line 231
```
seems to be coming from a `preg_match` that doesnt check if it yields nothing
```php
preg_match('/([^$&.]+)(?:((?:\.\.\.)?(?:\$|&)[^\s]+)(?:(\s+)(.*))?)?/', $tokens[($tag + 2)]['content'], $matches);

$typeLen   = strlen($matches[1]);
$type      = trim($matches[1]);
```

**Comment by gsherwood** [comment]
Tracked down to this function comment style:
```php
/**
 * Function comment.
 *
 * @param $bar
 *   Comment here.
 * @param ...
 *   Additional arguments here.
 *
 * @return
 *   Return value
 *
 */
function foo($bar) {
}
```

**Comment by gsherwood** [comment]
The problem is caused by the use of `...` instead of `...$paramName`. I didn't bother adding any specific error messages for this format. I'm sure Drupal doesn't use thee sniffs and so wouldn't care.

Thanks for reporting the bug.
